paasta_tools/deployd/watchers.py

```diff
@@ -284,10 +284,19 @@ def log(self):
         name = '.'.join([__name__, type(self).__name__])
         return logging.getLogger(name)
 
-    def filter_event(self, event):
+    def get_service_name_from_event(self, event):
+        """Get service_name from the file inotify event,
+        returns None if it is not an event we're interested in"""
         starts_with = ['marathon-', 'deployments.json']
         if any([event.name.startswith(x) for x in starts_with]):
-            return event
+            service_name = event.path.split('/')[-1]
```

Comment:
This is dangerous. What if the path doesn't have a '/' ?

You need to catch that

Reply:
I haven't changed this just moved it around. I'm pretty sure the paths are guaranteed to be absolute because we pass an absolute path to watch.